### PR TITLE
Iconify/minimize windows

### PR DESCRIPTION
**Pull request by rpc-scandinavia**
I discovered that it only is possible to maximize the window to full screen.
I implemented the "Ifonify" method and the "WindowState" property.
I am not able to test the Mac code, but it compiles on my Linux with MonoDevelop and a copy of "MonoMac.dll".

### CODE REVIEW

File: Xwt/Xwt.Backends/IWindowFrameBackend.cs [hunk 1]
```diff
@@ -82,10 +82,22 @@ public interface IWindowFrameBackend: IBackend
...
+		/// Gets or sets a value indicating whether this window is iconified (true) or normal (false).
+		/// </summary>
+		/// <value><c>true</c> if the window is iconified; otherwise, <c>false</c>.</value>
+		bool Iconify { get; set; }
+
+		/// <summary>
+		/// Gets or sets a value indicating whether this window is in full screen mode (true) and normal (false).
 		/// </summary>
 		/// <value><c>true</c> if the window is in full screen mode; otherwise, <c>false</c>.</value>
 		bool FullScreen { get; set; }
```
Comment (slluis) [inline Xwt/Xwt.Backends/IWindowFrameBackend.cs]: If we add the WindowState property, the Iconify and FullScreen properties are redundant in the backend, they can be removed.

File: Xwt/Xwt.Backends/IWindowFrameBackend.cs [hunk 2]
```diff
@@ -82,10 +82,22 @@ public interface IWindowFrameBackend: IBackend
 		bool Close ();
 
 		/// <summary>
-		/// Gets or sets a value indicating whether this window is in full screen mode
+		/// Gets or sets a value indicating whether this window is iconified (true) or normal (false).
+		/// </summary>
+		/// <value><c>true</c> if the window is iconified; otherwise, <c>false</c>.</value>
+		bool Iconify { get; set; }
```
Comment (slluis) [inline Xwt/Xwt.Backends/IWindowFrameBackend.cs]: This should be "Iconified"

File: Xwt/Xwt/WindowFrame.cs [hunk 3]
```diff
@@ -271,6 +271,15 @@ protected override BackendHost CreateBackendHost ()
 		}
 		
 		/// <summary>
+		/// Gets or sets a value indicating whether this window is iconified (true) or normal (false).
+		/// </summary>
+		/// <value><c>true</c> if the window is iconified; otherwise, <c>false</c>.</value>
+		public bool Iconify {
```
Comment (slluis) [inline Xwt/Xwt/WindowFrame.cs]: This should be "Iconified"

**Comment by sevoku** [comment]
This needs a rebase and maybe more descriptive commit messages.

**Comment by rpc-scandinavia** [comment]
Okay, what do you want me to do ?
What is a "rebase" ?

As I understand it, also from the comment by @slluis - I should remove the Iconify and FullScreen properties, and write a longer description.

**Comment by sevoku** [comment]
Yes :smile:
And if you look at the PR history, there are 6 commits with the same commit message. you should squash them into one single commit, or give the commits more descriptive messages. This can be done by soft-resetting git to 67a15a9 and recommitting (simple), or by (better) rebasing using -i and replacing "pick" by "squash" (look for "INTERACTIVE MODE" in http://git-scm.com/docs/git-rebase for more info about interactive rebasing) after pulling in the current master (will force you to resolve conflicts). 

And more general info about rebase: http://git-scm.com/book/en/v2/Git-Branching-Rebasing.

**Comment by sevoku** [comment]
I've rebased and optimized the implementation in #549. Thanks! Closing here.

**Pull request closed without merging**